modules/gitrepo/tree_path.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package gitrepo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/modules/git"
+)
+
+func GetTreePathLatestCommit(ctx context.Context, repo Repository, refName, treePath string) (*git.Commit, error) {
```

Comment:
This GetTreePathLatestCommit doesn't seem to make sense.

Router handlers could just call

```
commit, err := ctx.Repo.GitRepo.GetTreePathLatestCommitID(...)
```

Reply:
2f05073